include/ignition/gazebo/comms/ICommsModel.hh

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_GAZEBO_ICOMMSMODEL_HH_
+#define IGNITION_GAZEBO_ICOMMSMODEL_HH_
+
+#include <algorithm>
+#include <memory>
+
+#include <sdf/sdf.hh>
+#include <ignition/common/Profiler.hh>
+#include "ignition/gazebo/comms/Broker.hh"
+#include "ignition/gazebo/config.hh"
+#include "ignition/gazebo/EntityComponentManager.hh"
+#include "ignition/gazebo/System.hh"
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace comms
+{
+  // Forward declarations.
+  class MsgManager;
+
+  /// \brief Abstract interface to define how the environment should handle
+  /// communication simulation. As an example, this class could be responsible
+  /// for handling dropouts, decay and packet collisions.
+  ///
+  /// The derived comms models can be configured with the following SDF
+  /// parameters:
+  ///
+  /// * Optional parameters:
+  /// <step_size> If defined this will allow the comms model to run at a
+  /// higher frequency than the physics engine. This is useful when dealing
+  /// with ranging. If the <step_size> is set larger than the physics engine dt
+  /// then the comms model step size will default to dt.
+  /// Note: for consistency it is adviced that the dt is a multiple of timestep.
+  /// Units are in seconds.
+  ///    <messages_topic>: Topic name where the broker receives all the incoming
```

Comment:
Is this a copy-paste leftover? I don't see `Configure()` reading any of these parameters...

Reply:
Removed in [1140b56](https://github.com/ignitionrobotics/ign-gazebo/pull/1416/commits/1140b565c175cbdae114a39158f4639a373b04db).